planning/behavior_path_planner/src/scene_module/start_planner/start_planner_module.cpp

```diff
@@ -211,25 +242,35 @@ bool StartPlannerModule::isMoving() const
          parameters_->th_stopped_velocity;
 }
 
-bool StartPlannerModule::isExecutionReady() const
+bool StartPlannerModule::isStopped()
 {
-  // when found_pull_out_path is false,the path is not generated and approval shouldn't be
-  // allowed
-  if (!status_.found_pull_out_path) {
-    RCLCPP_ERROR_THROTTLE(getLogger(), *clock_, 5000, "Pull over path is not found");
-    return false;
-  }
-
-  if (
-    parameters_->safety_check_params.enable_safety_check && status_.driving_forward &&
-    isWaitingApproval()) {
-    if (!isSafePath()) {
-      RCLCPP_ERROR_THROTTLE(getLogger(), *clock_, 5000, "Path is not safe against dynamic objects");
-      stop_pose_ = planner_data_->self_odometry->pose.pose;
-      return false;
+  odometry_buffer_.push_back(planner_data_->self_odometry);
+  // Delete old data in buffer
+  while (rclcpp::ok()) {
+    const auto time_diff = rclcpp::Time(odometry_buffer_.back()->header.stamp) -
+                           rclcpp::Time(odometry_buffer_.front()->header.stamp);
+    if (time_diff.seconds() < parameters_->th_stopped_time) {
+      break;
     }
+    odometry_buffer_.pop_front();
   }
-  return true;
+  return !std::any_of(
+    odometry_buffer_.begin(), odometry_buffer_.end(), [this](const auto & odometry) {
+      return utils::l2Norm(odometry->twist.twist.linear) > parameters_->th_stopped_velocity;
+    });
+}
+
+bool StartPlannerModule::isExecutionReady() const
+{
+  // Evaluate safety. The situation is not safe if any of the following conditions are met:
+  // 1. pull out path has not been found
+  // 2. waiting for approval and there is a collision with dynamic objects
+  const bool is_safe =
+    status_.found_pull_out_path && (!isWaitingApproval() || checkCollisionWithDynamicObjects());
```

Comment:
changed as follows
```
  // Evaluate safety. The situation is not safe if any of the following conditions are met:
  // 1. pull out path has not been found
  // 2. waiting for approval and there is a collision with dynamic objects
  if (!status_.found_pull_out_path) {
    is_safe = false;
  }

  if (requiresDynamicObjectsCollisionDetection()) {
    is_safe = !hasCollisionWithDynamicObjects();
  }
  ```

Reply:
why not just `is_safe = isSafePath();` ? :pray:

Reply:
`hasCollisionWithDynamicObjects` is currently just `!isSafePath();` but will be updated as TODO in other PR?

```
bool StartPlannerModule::hasCollisionWithDynamicObjects() const
{
  // TODO(Sugahara): update path, params for predicted path and so on in this function to avoid
  // mutable
  return !isSafePath();
}
```

Reply:
> asCollisionWithDynamicObjects is currently just !isSafePath(); but will be updated as TODO in other PR?

Yes  I mean that.
want to separate object filetering part and create predicted path part from `isSafe()` function